x-pack/plugins/security/server/authorization/validate_es_response.ts

```diff
@@ -9,10 +9,13 @@ import { schema } from '@kbn/config-schema';
 
 import type { HasPrivilegesResponse } from './types';
 
-const anyBoolean = schema.boolean();
-const anyBooleanArray = schema.arrayOf(anyBoolean);
-const anyString = schema.string();
-const anyObject = schema.object({}, { unknowns: 'allow' });
+const baseResponseSchema = schema.object({
+  username: schema.string(),
+  has_all_requested: schema.boolean(),
+  application: schema.object({}, { unknowns: 'allow' }),
+  cluster: schema.object({}, { unknowns: 'allow' }),
+  index: schema.object({}, { unknowns: 'allow' }),
+});
```

Comment:
__note (to myself):__ Looks like we can eventually get rid of `@kbn/config-schema` completely here since you did the hardest part now :slightly_smiling_face:

Reply:
The `schema.validate` part still consumes a lot, so yeah it would be useful